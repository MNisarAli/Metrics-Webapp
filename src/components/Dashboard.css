.dashboard {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem 1rem;
}

.dashboard-map {
  opacity: 0.35;
  min-height: 10rem;
  max-height: 14rem;
  width: 50%;
}

.title-container {
  align-items: flex-start;
  text-align: right;
  max-width: 50%;
}

.page-title {
  font-size: 1.6rem;
  text-transform: uppercase;
}
