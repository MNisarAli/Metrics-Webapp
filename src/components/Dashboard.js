import PropTypes from 'prop-types';
import './Dashboard.css';

const Dashboard = ({ map, name, population }) => (
  <section className="dashboard">
    <img src={map} alt={name} className="dashboard-map" />
    <article className="title-container">
      <h2 className="page-title">{name}</h2>
      <span>{population}</span>
    </article>
  </section>
);

Dashboard.propTypes = {
  name: PropTypes.string,
  population: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  map: PropTypes.string,
};

Dashboard.defaultProps = {
  name: '',
  population: '',
  map: '',
};

export default Dashboard;
