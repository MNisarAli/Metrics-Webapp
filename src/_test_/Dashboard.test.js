import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import Dashboard from '../components/Dashboard';

describe('Dashboard component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Dashboard />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the Dashboard component with non-default props', () => {
    const { getByAltText, getByText } = render(
      <Dashboard name="Test City" population={100000} map="test-map.png" />,
    );
    const pageImage = getByAltText('Test City');
    const pageTitle = getByText('Test City');
    expect(pageImage).toBeInTheDocument();
    expect(pageTitle).toBeInTheDocument();
  });

  test('renders the correct population in the Dashboard component', () => {
    const { getByText } = render(
      <Dashboard name="Test City" population={100000} map="test-map.png" />,
    );
    const population = getByText('100000');
    expect(population).toBeInTheDocument();
  });
});
