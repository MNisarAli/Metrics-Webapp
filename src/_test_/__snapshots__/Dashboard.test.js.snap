// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly 1`] = `
<section
  className="dashboard"
>
  <img
    alt=""
    className="dashboard-map"
    src=""
  />
  <article
    className="title-container"
  >
    <h2
      className="page-title"
    />
    <span />
  </article>
</section>
`;
